### PR TITLE
fix(yarn) pnp workspace support

### PR DESCRIPTION
**Pull request by mufasa71**
Yarn berry works great with single mode, but does not works if inside workspace directory. I found that this line does not respect root monorepo:
https://github.com/volta-cli/volta/blob/7827adefcb57ffe3beb7896c7ed483b24e619556/crates/volta-core/src/project/mod.rs#L176

as it expect `.pnp.js` on same level as root dir. But workspace dir does not have `.pnp.js` hence it works fine with yarn berry. like `yarn why packagename`

### CODE REVIEW

File: crates/volta-core/src/project/mod.rs
Comment (charlespierce) [inline crates/volta-core/src/project/mod.rs:183]: I think this winds up duplicating work: `self.workspace_roots()` includes as the first entry the parent of `self.manifest_file`, so we're re-checking that. Instead, I think this check that you have is already exactly what we want (and can replace the whole `self.manifest_file...` check)

```rust
self
    .workspace_roots()
    .any(|dir| dir.join(".pnp.cjs").exists() || dir.join(".pnp.js").exists())
```

**Review by charlespierce — CHANGES_REQUESTED** [review]
Thanks for taking this on! A suggestion to avoid duplicating work. Also, there may be some merge conflicts with #1388 which is adding another file to the checks here, so if that is merged first, it will need to be resolved.

File: crates/volta-core/src/project/mod.rs
Comment (mufasa71) [reply crates/volta-core/src/project/mod.rs]: Thanks I applied changes, will wait until #1388 is merged to resolve conflicts.

Comment (mufasa71) [reply crates/volta-core/src/project/mod.rs]: Looks like #1388 is merged, can we merge this now?

**Review by charlespierce — APPROVED** [review]

**Pull request merged**
